pkg/deploy/api/v1beta3/types.go

```diff
@@ -89,6 +89,9 @@ type LifecycleHook struct {
 	// FailurePolicy specifies what action to take if the hook fails.
 	FailurePolicy LifecycleHookFailurePolicy `json:"failurePolicy" description:"what action to take if the hook fails"`
 
+	// MaxRestarts specifies a number of allowed lifecycle pod container restarts
+	// before we treat the lifecycle hook as failed.
+	MaxRestarts int `json:"maxRestarts,omitempty" description:"a number of allowed container restarts before the lifecycle hook is considered as failed"`
```

Comment:
Should be int32 or int64. Upstream they are using the former for fields like this and the later for counters. So I would say int32